### PR TITLE
(DOCSP-23965): RQL list query comparisons

### PR DESCRIPTION
**Pull request by nlarew**
### Jira

- (DOCSP-23965): RQL list query comparisons

### Staged Changes (Requires MongoDB Corp SSO)

- [Realm Query Language](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/rql-lists/realm-query-language/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/realm-query-language.txt [hunk 1]
```diff
@@ -718,6 +719,83 @@ If the predicate returns true, the object is included in the output collection.
...
+
+List Comparisons
+----------------
+
+You can use :ref:`comparison operators <rql-comparison-operators>` and
+:ref:`collection operators <rql-collection-operators>` to filter based
+on lists of data. You can compare any type of list, including
+collections of Realm objects, lists defined in the query and native list
+objects passed in a :ref:`parameterized expression
+<rql-parameterized-queries>`.
```
Comment (mongodben) [inline source/realm-query-language.txt:732]: generally lgtm. but could you add a sentence or 2 on why user might want to do this? 

i remember there was some flexible sync use case that made this very compelling.

relatedly, is there anything that should be added to the `Flexible Sync RQL Limitations` section per these changes?

**Comment by nlarew** [comment]
@mongodben Mind taking another look? I added some examples with tests+bluehawk & added a short note to flex sync limitations

File: source/realm-query-language.txt [hunk 2]
```diff
@@ -718,6 +719,101 @@ If the predicate returns true, the object is included in the output collection.
...
+You can compare any type of valid list. This includes:
+
+- collections of Realm objects, which let you filter against other data
+  in the realm.
+
+  .. literalinclude:: /examples/generated/realm-query-language/realm-query-language.snippet.list-comparisons-collection.js
+     :language: js
+
+- lists defined directly in the query, which let you filter against
+  static data.
```
Comment (mongodben) [inline source/realm-query-language.txt:740]: can you add sentence here abt how you should use `{}` to define a list?

**Review by mongodben — APPROVED** [review]
lgtm, with one requested addition

**Pull request merged**